src/utils/totp.ts

```diff
@@ -0,0 +1,100 @@
+import { HD, PublicKey } from '@bsv/sdk';
+import { Contact } from '../types';
+import { SpvWalletClient } from '../client';
+import { base32 } from "@scure/base"
+
+import { totp } from 'otplib';
+
+export const DEFAULT_TOTP_PERIOD = 30;
+export const DEFAULT_TOTP_DIGITS = 2;
+
+/**
+ * Generates a TOTP for a given contact
+ *
+ * @param client - The SpvWalletClient
+ * @param contact - The Contact
+ * @param period - The TOTP period (default: 30)
+ * @param digits - The number of TOTP digits (default: 2)
+ * @returns The generated TOTP as a string
+ */
+export const GenerateTotpForContact = (
+  client: SpvWalletClient,
+  contact: Contact,
+  period: number = DEFAULT_TOTP_PERIOD,
+  digits: number = DEFAULT_TOTP_DIGITS,
+): string => {
+  const sharedSecret: string = makeSharedSecret(contact, client);
+
+  totp.options = {digits: digits, step: period, epoch: Date.now()}
+
+  let secret = directedSecret(sharedSecret, contact.paymail);
+  return totp.generate(secret);
```

Comment:
I think, it is unsafe to do it this way (`totp.options = ...`)
Because of the nature of the `otplib`, we modifies here "global" options witch causes several issues:
- if a "user" of our `js-client` use the `totp` outside of "our" code - he will share the same options with js-client which may lead to undefined behaviour
- As creators of this lib say: "use epoch witch caution". Here we globally set the time base for all... I think we don't have to set it at all, because it set under-the-hood here: https://github.com/yeojz/otplib/blob/147e67637377196e3c59cf93b12ac325aca76fac/packages/otplib-core/src/totp.ts#L149
![image](https://github.com/bitcoin-sv/spv-wallet-js-client/assets/152964795/d303d284-3770-4921-88a9-423116974220)

I can see following solutions:
- use otplib/core and create an instance of TOTP every time it is needed (with specialized options)
- use `clone()` before setting the options, like that:
```typescript
let secret = directedSecret(sharedSecret, contact.paymail);

  const totpInstance = totp.clone()
  totpInstance.options = {digits: digits, step: period}
  totpInstance.generate(secret)
```